compiler-plugin/src/main/java/io/ballerina/lib/ai/np/compilerplugin/PromptGenerator.java

```diff
@@ -0,0 +1,271 @@
+package io.ballerina.lib.ai.np.compilerplugin;
+
+import com.google.gson.JsonArray;
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.ConstantSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.InterpolationNode;
+import io.ballerina.compiler.syntax.tree.LiteralValueToken;
+import io.ballerina.compiler.syntax.tree.NaturalExpressionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+
+import java.io.IOException;
+
+import static io.ballerina.lib.ai.np.compilerplugin.CommonUtils.CONTENT;
+import static io.ballerina.lib.ai.np.compilerplugin.CommonUtils.FILE_PATH;
+import static io.ballerina.lib.ai.np.compilerplugin.CommonUtils.retrieveLangLibs;
+
+public class PromptGenerator {
+    public static final String LANG_LIBS_PATH = "/langlibs.json";
+    private static final String LANG_LIBS;
+    static {
+        try {
+            LANG_LIBS = retrieveLangLibs(LANG_LIBS_PATH);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String getSystemPromptPrefix() {
+        return """
+                You are an expert assistant who specializes in writing Ballerina code.
+                Your goal is to ONLY answer Ballerina related queries. You should always answer with
+                    accurate and functional Ballerina code that addresses the specified query while adhering
+                    to the constraints.
+                You will be provided with following inputs:
+           """;
+    }
+
+    public static String getSystemPromptSuffix() {
+        return """
+                Langlibs
+                <langlibs>
+                   %s
+                </langlibs>
```

Comment:
Did we try without this? Or was it a direct duplication of what we have already? I think we should explore refining the prompts too, given that the current models are better at generating Ballerina code.

Reply:
That part is not completed yet